experimental/HasseSchmidt/src/HasseSchmidt.jl

```diff
@@ -0,0 +1,162 @@
+export hasse_derivatives
+
+### We consider Hasse-Schmidt derivatives of polynomials as seen in
+###
+###   	[FRS21](@cite) Fruehbis-Krueger, Ristau, Schober: 'Embedded desingularization for arithmetic surfaces -- toward a parallel implementation'
+###			
+### This is a special case of a more general definition of a Hasse-Schmidt derivative. These more general and rigorous definitions can be found in the following sources:
+###
+###			[Cut04](@cite) Cutkosky: 'Resolution of Singularities'
+###			[Haz12](@cite) Michiel Hazewinkel: 'Hasse-Schmidt derivations and the Hopf algebra of noncommutative symmetric functions'
+###
+
+################################################################################
+### HASSE-SCHMIDT derivatives for single polynomials
+
+@doc raw"""
+    hasse_derivatives(f::MPolyRingElem)
+
+Return a list of Hasse-Schmidt derivatives of `f`, each with a multiindex `[a_1, ..., a_n]`, where `a_i` describes the number of times `f` was derived w.r.t. the `i`-th variable. 
+
+Hasse-Schmidt derivatives as seen in [FRS21](@cite).
+For more general and rigorous definition see [Cut04](@cite) or [Haz12](@cite).
+
+# Examples
+```jldoctest
+julia> R, (x, y) = polynomial_ring(ZZ, ["x", "y"]);
+
+julia> f = 5*x^2 + 3*y^5;
+
+julia> hasse_derivatives(f)
+8-element Vector{Vector{Any}}:
+ [[0, 5], 3]
+ [[0, 4], 15*y]
+ [[0, 3], 30*y^2]
+ [[2, 0], 5]
+ [[0, 2], 30*y^3]
+ [[1, 0], 10*x]
+ [[0, 1], 15*y^4]
+ [[0, 0], 5*x^2 + 3*y^5]
+```
+"""
+function hasse_derivatives(f::MPolyRingElem)
+  R = parent(f)
+  # x = gens(R)
+  # n = ngens(R)
```

Comment:
please remove these two lines. They seem to be leftovers from previous testing or debugging.